java-spiffe-core/src/main/proto/workload.proto

```diff
@@ -39,16 +40,56 @@ message X509SVID {
     // CA certificates belonging to the Trust Domain
     // ASN.1 DER encoded
     bytes bundle = 4;
+}
+
+message JWTSVID {
+    string spiffe_id = 1;
+
+    // Encoded using JWS Compact Serialization
+    string svid = 2;
+}
 
-    // List of trust domains the SVID federates with, which corresponds to
-    // keys in the federated_bundles map in the X509SVIDResponse message.
-    repeated string federates_with = 5;
+message JWTSVIDRequest {
+    repeated string audience = 1;
+
+    // SPIFFE ID of the JWT being requested
+    // If not set, all IDs will be returned
+    string spiffe_id = 2;
+}
+
+message JWTSVIDResponse {
+    repeated JWTSVID svids = 1;
+}
+
+message JWTBundlesRequest { }
+
+message JWTBundlesResponse {
+    // JWK sets, keyed by trust domain URI
+    map<string, bytes> bundles = 1;
+}
+
+message ValidateJWTSVIDRequest {
+    string audience = 1;
+
+    // Encoded using JWS Compact Serialization
+    string svid = 2;
+}
+
+message ValidateJWTSVIDResponse {
+    string spiffe_id = 1;
+    google.protobuf.Struct claims = 2;
 }
 
 service SpiffeWorkloadAPI {
```

Comment:
It would be ideal to not have to redefine the Workload API contract in each language's SPIFFE library. Are there any plans to maintain the .proto file as a versioned artifact external to each library repository?

Reply:
Well, single source of truth for protobuf definitions is a debated topic. I prefer to update it manually in each library repository  as part of making the code changes that the new proto definition possibly requires.